handler/core/src/main/java/io/knotx/fragments/task/options/GraphNodeOptions.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.task.options;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * It is {@link io.knotx.fragments.engine.Task} processing configuration. Task is graph of nodes (in
+ * fact it is tree structure). It defines {@link NodeOptions} and outgoing directed graph edges,
+ * called {@code Transitions}.
+ *
+ * It represents JSON configuration:
+ * <pre>
+ * {
+ *   node = {
+ *     factory = action
+ *     config {
+ *       action = a
+ *     }
+ *   }
+ *   onTransitions {
+ *     _success {
+ *       node = {
+ *         factory = action
+ *         config {
+ *           action = b
+ *         }
+ *       }
+ *     }
+ *   }
+ * }
+ * </pre>
+ *
+ * Please note that Transitions define next graph nodes.
+ */
+@DataObject(generateConverter = true)
+public class GraphNodeOptions {
```

Comment:
Since we are messing with configuration structure, maybe it would be a good time to get rid of `Options` postfix here and in other places for this module? It does not help to understand the model of a graph.

Reply:
Not sure we want to do this. We have 3 modules now:
- fragments handler . (`ActionFactory`, `TaskProvider`, `NodeProvider`)
- task engine (Task, `ActionNode` -> `SingleNode`, `CompositeNode`)
`NodeProvider` creates `SingleNode` or `CompositeNode` based on its implementation, finally we create `Task`. From task engine point of view, there are no Actions at all. There is function to execute: `Function<FragmentContext, Single<FragmentResult>>`.

JSON configuration should allow to build different Tasks configurations, but it do not directly create `SingleNode` and `CompositeNode`. However, `ActionNodeProviderFactory` will create `SingleNode`.

This is the abstraction between handler and engine. Engine can be used from other modules.